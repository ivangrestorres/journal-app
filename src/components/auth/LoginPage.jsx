import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { startGoogleLogin, startLoginEmailPassword } from "../../actions/auth";
import { useForm } from "../../hooks/useForm";

export const LoginPage = () => {
    const dispatch = useDispatch();
    const { loading, msgLoginError } = useSelector((state) => state.ui);

    const initialForm = {
        email: "kk@gmail.com",
        password: "12345aA",
    };
    const [formValues, handleInputChange] = useForm(initialForm);

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(startLoginEmailPassword(email, password));
    };

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    };

    return (
        <div>
            <h3 className={"auth_title"}>Login</h3>
            {msgLoginError && (
                <div className={"auth_alert-error"}>{msgLoginError}</div>
            )}
            <form onSubmit={handleLogin}>
                <input
                    type="text"
                    placeholder={"Email"}
                    name={"email"}
                    className={"auth_input"}
                    autoComplete={"off"}
                    value={email}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder={"Password"}
                    name={"password"}
                    className={"auth_input"}
                    value={password}
                    onChange={handleInputChange}
                />
                <button
                    className={"btn btn-primary btn-block"}
                    type={"submit"}
                    disabled={loading}
                >
                    Login
                </button>

                <div className={"auth_social-networks"}>
                    <p>Login with social networks</p>
                    <div className="google-btn" onClick={handleGoogleLogin}>
                        <div className="google-icon-wrapper">
                            <img
                                className="google-icon"
                                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                                alt="google button"
                            />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>
                <Link className={"link"} to={"/auth/register"}>
                    Create new account
                </Link>
            </form>
        </div>
    );
};
