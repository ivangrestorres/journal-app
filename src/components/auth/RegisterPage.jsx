import React from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import validator from "validator";
import { useDispatch, useSelector } from "react-redux";
import { startRegisterWithEmailPasswordName } from "../../actions/auth";
import Swal from "sweetalert2";

export const RegisterPage = () => {
    const dispatch = useDispatch();
    const { msgError } = useSelector((state) => state.ui);

    const initialForm = {
        name: "",
        email: "",
        password: "",
        password2: "",
    };

    const [formValues, handleInputChange] = useForm(initialForm);
    const { name, email, password, password2 } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        if (isFormValid()) {
            dispatch(startRegisterWithEmailPasswordName(email, password, name));
        }
    };

    const isFormValid = () => {
        if (name.trim().length === 0) {
            Swal.fire("Fail", "Name incorrect", "error");
            return false;
        }
        if (!validator.isEmail(email)) {
            Swal.fire("Fail", "Email incorrect", "error");
            return false;
        }
        if (password !== password2 || password.length < 5) {
            Swal.fire("Fail", "Password incorrect", "error");
            return false;
        }
        return true;
    };

    return (
        <div>
            <h3 className={"auth_title"}>Register</h3>
            <form
                onSubmit={handleRegister}
                className={"animate__animated animate__fadeIn animate__faster"}
            >
                {msgError && (
                    <div className={"auth_alert-error"}>{msgError}</div>
                )}
                <input
                    type="text"
                    placeholder={"Name"}
                    name={"name"}
                    value={name}
                    className={"auth_input"}
                    autoComplete={"off"}
                    onChange={handleInputChange}
                />
                <input
                    type="text"
                    placeholder={"Email"}
                    name={"email"}
                    value={email}
                    className={"auth_input"}
                    autoComplete={"off"}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder={"Password"}
                    name={"password"}
                    value={password}
                    className={"auth_input"}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder={"Confirm password"}
                    name={"password2"}
                    value={password2}
                    className={"auth_input"}
                    onChange={handleInputChange}
                />
                <button
                    className={"btn btn-primary btn-block mt-5 mb-5"}
                    type={"submit"}
                >
                    Register
                </button>
                <Link className={"link"} to={"/auth/login"}>
                    Already register?
                </Link>
            </form>
        </div>
    );
};
