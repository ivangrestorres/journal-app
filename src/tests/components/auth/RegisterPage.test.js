import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom";

import { MemoryRouter } from "react-router";
import { RegisterPage } from "../../../components/auth/RegisterPage";
import { startRegisterWithEmailPasswordName } from "../../../actions/auth";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock("../../../actions/auth", () => ({
    startRegisterWithEmailPasswordName: jest.fn(),
}));

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        notes: [],
        active: null,
    },
};
let store = mockStore(initState);
store.dispatch = jest.fn();

describe("Pruebas <RegisterPage />", () => {
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
                <RegisterPage />
            </MemoryRouter>
        </Provider>
    );

    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de disparar la accion de startRegisterWithEmailPasswordName", () => {
        wrapper.find("form").prop("onSubmit")({ preventDefault() {} });
        expect(startRegisterWithEmailPasswordName).not.toHaveBeenCalled();

        wrapper.find("input[name='name']").simulate("change", {
            target: { value: "Ivan", name: "name" },
        });

        wrapper.find("input[name='email']").simulate("change", {
            target: { value: "a@gmail.com", name: "email" },
        });

        wrapper.find("input[name='email']").simulate("change", {
            target: { value: "12345aA", name: "password" },
        });

        wrapper.find("input[name='email']").simulate("change", {
            target: { value: "12345aA", name: "password2" },
        });

        wrapper.find("form").prop("onSubmit")({ preventDefault() {} });
        expect(startRegisterWithEmailPasswordName).toHaveBeenLastCalledWith(
            "a@gmail.com",
            "12345aA",
            "Ivan"
        );
    });
});
