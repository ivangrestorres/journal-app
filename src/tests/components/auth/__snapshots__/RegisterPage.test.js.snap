// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <RegisterPage /> debe de mostrarse correctamente 1`] = `
<div>
  <h3
    className="auth_title"
  >
    Register
  </h3>
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth_input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth_input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      className="auth_input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="auth_input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block mt-5 mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already register?
    </a>
  </form>
</div>
`;
